scripts/actions/check-job-progress.js

```diff
@@ -100,6 +100,88 @@ const getBatchStatus = async ({ batchUid, jobId }) => {
   }
 };
 
+/**
+ * @param {Object[]} slugStatuses
+ * @param {Boolean} slugStatuses[].ok - Boolean flag signaling if translation deserialized. If it did, we can treat this translation as having completed.
+ * @param {String} slugStatuses[].slug - Slug representing file path translated.
+ * @param {String} slugStatuses[].jobId - Job id translation is associated with.
+ */
+const aggregateStatuses = (slugStatuses) => {
+  let totalFailures = 0;
+  let totalSuccesses = 0;
+  const jobStatuses = {};
+
+  slugStatuses.map(({ ok, jobId }) => {
+    if (!(jobId in jobStatuses)) {
+      jobStatuses[jobId] = { successes: 0, failures: 0 };
+    }
+
+    if (ok) {
+      totalSuccesses += 1;
+      jobStatuses[jobId].successes += 1;
+    } else {
+      totalFailures += 1;
+      jobStatuses[jobId].failures += 1;
+    }
+  });
+
+  return { totalFailures, totalSuccesses, jobStatuses };
+};
+
+/**
+ * @param {Object[]} slugStatuses
+ * @param {Boolean} slugStatuses[].ok - Boolean flag signaling if translation deserialized. If it did, we can treat this translation as having completed.
+ * @param {String} slugStatuses[].slug - Slug representing file path translated.
+ * @param {String} slugStatuses[].jobId - Job id translation is associated with.
+ */
+const updateTranslationRecords = async (slugStatuses) => {
+  // TODO: need to update this when we implement multiple locales. This only works for one locale.
+
+  await Promise.all(
+    slugStatuses.map(async ({ ok, slug }) => {
+      let updateStatus = ok ? 'COMPLETED' : 'ERRORED';
+
+      const record = await updateTranslations(
+        { slug, status: 'IN_PROGRESS' },
+        { status: updateStatus }
+      );
+
+      console.log(`Translation ${record.id} marked as ${updateStatus}`);
+    })
+  );
+};
+
+/**
+ * @param {Object.<string, Object.<string, number>} jobStatuses - object whose top level keys are job ids, and whose values are { succeses: x, failures: y }, which describe failure and success count for that given job.
```

Comment:
We could consider using a [JSDoc `@typedef`](https://jsdoc.app/tags-typedef.html) to define the shape of these status objects (success & failure counts).